### PR TITLE
skip-release: Add workflow to auto-update versions

### PR DESCRIPTION
**Pull request by SuperQ**
Add a script and workflow to automatically check for upstream releases and create PRs to update the associated role.

### CODE REVIEW

**Comment by SuperQ** [comment]
This is a port over the from the old auto-maintence repo.

**Review by gardar — COMMENTED** [review]
The script needs to be updated to bump the default version in the argument_specs as well.

But since there is also in other places where we need to think about keeping versions up to date, perhaps we should look into using renovatebot/dependabot or similar?
I'm talking about the versions for test-requirements, github actions in the workflows, ansible collections we depend on, ansible itself, and the linux distros that the collection supports.

**Comment by SuperQ** [comment]
Yes, adding some dependabot stuff would be nice. But I don't think there's anything off-the-shelf that can deal with these role/repo specific updates.

**Comment by SuperQ** [comment]
Test PR: https://github.com/prometheus-community/ansible/pull/59

**Comment by gardar** [comment]
> Yes, adding some dependabot stuff would be nice. But I don't think there's anything off-the-shelf that can deal with these role/repo specific updates.

I actually think renovatebot comes close, it supports `ansible galaxy`, `github releases`, `github workflow actions`, `pypi`.. Only thing missing would be linux distros but that's a low hanging fruit.

https://docs.renovatebot.com/modules/datasource/
https://docs.renovatebot.com/modules/manager/

**Comment by paulfantom** [comment]
I agree with @gardar and also think that renovate may be a way forward here. It will allow us to manage less things related to credentials and also use a standard component instead of artisian scripts.

**Comment by SuperQ** [comment]
Yup, just had to fix a shell linter issue.

**Review by gardar — APPROVED** [review]

File: .github/workflows/version_bumper.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          - repo: "prometheus/alertmanager"
+            role: "alertmanager"
+          - repo: "prometheus/blackbox_exporter"
+            role: "blackbox_exporter"
+          - repo: "prometheus/node_exporter"
+            role: "node_exporter"
+          - repo: "prometheus/prometheus"
+            role: "prometheus"
+          - repo: "prometheus/snmp_exporter"
+            role: "snmp_exporter"
```
Comment (gardar) [inline .github/workflows/version_bumper.yml:27]: Can we do the matrix without hard-coding it? Like we do for the molecule tests

Comment (SuperQ) [reply .github/workflows/version_bumper.yml:27]: I don't know if there's a sane way to do that, what if the repo is `prometheus-community/postgres_exporter` and the role is `postgres_exporter`?

**Pull request merged**